lib/traits/thermostat_widget.dart

```diff
@@ -66,16 +66,18 @@ class ThermostatWidget extends StatelessWidget with ToastNotifications {
     }
     final isCool = currentMode == AvailableThermostatMode.COOL;
     final temperatureRange = isCool
-        ? _thermostatProvider.getCoolTemperatureRange!
-        : _thermostatProvider.getHeatTemperatureRange!;
+        ? _thermostatProvider.getCoolTemperatureRange
+        : _thermostatProvider.getHeatTemperatureRange;
     double value = _thermostatProvider.getTargetTemperatureState!;
 
-    if ((value < temperatureRange.min)) {
-      value = temperatureRange.min;
-    }
+    if (temperatureRange != null) {
```

Comment:
TemperatureRange should not be null if it is HEAT/COOL mode so it may be safe to just throw an exception here or show error message if that is the case. I think we will have to add exclusive `isHeat` boolean, otherwise this is a bug actually. I recalled at one time to do that but later missed it.

Reply:
According to the documentation the range is nullable and there shouldn't be validation done if it is.
<img width="869" alt="Screen Shot 2022-02-23 at 10 25 31 AM" src="https://user-images.githubusercontent.com/4786115/155350522-95cd3d2c-7502-4ae5-8d3f-6f496314a4f3.png">